src/Categories/Double/Core.agda

```diff
@@ -0,0 +1,315 @@
+{-# OPTIONS --without-K --safe #-}
+module Categories.Double.Core where
+
+open import Level
+
+open import Relation.Binary using (Rel; IsEquivalence; Setoid; REL)
+open import Relation.Binary.PropositionalEquality as ≡ using (_≡_)
+import Relation.Binary.Reasoning.Setoid as SetoidR
+
+import Categories.Category.Unbundled as Cat
+
+{-
+When defining a double category over a setoid enriched category
+we wish to state an identity law like the following:
+```
+
+          f
+      T₁  →  T₂
+    v ↓   s  ↓ w
+      B₁  →  B₂              f
+          g                T₁ → T₂
+          ∘        ≃    v  ↓    ↓ w
+          h                B₁ → B₂
+      B₁  →  B₂             h ∘ g
+   id ↓  id  ↓ id
+      B₁  →  B₂
+          h
+```
+
+Note that if we try to define homogeneous equality between squares of
+the form: `Sq₂ f (h ∘ g) v w`, then this will not work as the left-hand
+verical composite is a square of type `Sq₂ f (h ∘ g) (id ∘ v) (id ∘ w)`.
+Instead, we define equality of squares to mean an equalities in each of
+the horizontal or vertical hom setoids.
+
+This equality of squares in a double category is captured in the
+definition |SqEqualtiy|.
+
+-}
+module _ {o ℓ ℓ' e e' : Level} {Obj : Set o}
+         (Hor : Cat.Category Obj ℓ e) (Ver : Cat.Category Obj ℓ' e') where
+  private
+    module H = Cat.Category Hor
+    module V = Cat.Category Ver
+    _≈ₕ_ : ∀ {A B} → Rel (A H.⇒ B) e
+    _≈ₕ_ = H._≈_
+    _≈ᵥ_ : ∀ {A B} → Rel (A V.⇒ B) e'
+    _≈ᵥ_ = V._≈_
+  record SqEquality
+    {T₁ T₂ B₁ B₂ : Obj}
```

Comment:
So now that the big things are over, we get to the most controversial part: naming!

I suggest that:
- you name the objects A B C D
- you name the arrows T, B, L and R (with subscript 1 and 2), i.e. drop the h and v prefix
- name the fields T≈, .., R≈.